Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -121,41 +131,50 @@
       "inputs": {
         "Service": "ec2",
         "Api": "CopySnapshot",
-        "SourceSnapshotId": "{{extractSnapshotID.SNAPSHOTID}}",
+        "SourceSnapshotId": "{{extractSnapshotId.SnapshotId}}",
         "SourceRegion": "{{global:REGION}}",
         "Encrypted": true,
-        "KmsKeyId": "{{KmsKeyId}}",
+        "KmsKeyId": "{{kmsKeyId}}",
         "DestinationRegion": "{{global:REGION}}"
       },
       "outputs": [
         {
-          "Name": "EncryptedSnapshotID",
+          "Name": "encryptedSnapshotId",
           "Selector": "$.SnapshotId",
           "Type": "String"
         }
       ],
-      "name": "CopySnapshot",
+      "name": "copyAndEncryptSnapshot",
       "action": "aws:executeAwsApi",
       "timeoutSeconds": 3600,
       "onFailure": "Abort"
     },
     {
+      "name": "waitForSnapshot",
+      "action": "aws:waitForAwsResourceProperty",
+      "timeoutSeconds": "180",
```

Comment:
This timeout seems fairly short. I would wait up to 30 minutes to cover most of the cases.
Even better, make this maxWait an optional input parameters (with a default we think is reasonable) so that the end user can decide how long he is willing to wait. If I am encrypting a huge root volume I have never snapshot'd, I may expect it to take a long time, even hours.

Reply:
Increased the timeout to 1800 which should be sufficient in the latest commit.  You'll also notice that I created two new parameters "maxWaitForCreateSnapshot" and "maxWaitForSnapshotWithEncryption" which are to be used to give users options.  However, whenever, I try to use those variables, I encounter weird error.  When I try to create a new Document, it always returns and says "JSON not well formed" and it'll always point to the line after the timeoutSeconds.  For example, if I have following code:
      "name": "waitForSnapshot",
      "action": "aws:waitForAwsResourceProperty",
      "timeoutSeconds": "{{maxWaitForSnapshotWithEncryption}}",
then it'll always point to the "action" line and never lets me create the new Document.  I change the "timeoutSeconds" to a real integer value and then it works.  Have you encountered anything like that before?